dotCMS/src/main/java/com/dotmarketing/velocity/ClientVelocityServlet.java

```diff
@@ -4,22 +4,75 @@
  */
 package com.dotmarketing.velocity;
 
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
+import com.dotcms.util.SecurityUtils;
+import com.dotmarketing.util.PageRequestModeUtil;
+import com.dotmarketing.util.URLEncoder;
+import org.apache.jasper.security.SecurityUtil;
+import org.apache.velocity.runtime.parser.node.BooleanPropertyExecutor;
 import org.apache.velocity.tools.view.context.ChainedContext;
 
 import com.dotmarketing.util.Config;
 
+import java.io.IOException;
+
 /**
  * @author maria
  */
 public class ClientVelocityServlet extends VelocityServlet {
 
+    private URLEncoder encoder = new URLEncoder();
+    private static String IN_FRAME_PARAMETER_NAME = "in_frame";
+
     /**
      * 
      */
     private static final long serialVersionUID = 1L;
 
+    protected void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
+        boolean ADMIN_MODE = PageRequestModeUtil.isAdminMode( req.getSession() );
+        boolean notRedirect = notRedirect( req );
+
+        if ( ADMIN_MODE  && !notRedirect ){
+            String pageUrl = req.getAttribute("javax.servlet.forward.request_uri").toString();
+
+            if (pageUrl != null && pageUrl.indexOf(IN_FRAME_PARAMETER_NAME) == -1) {
+                String redirectURL = getRedirectURL(pageUrl);
+                response.sendRedirect(redirectURL);
+                return;
+            }
+        }
+
+        super.service(req, response);
+    }
+
+    public String getRedirectURL(String pageUrl){
+        return String.format("/dotAdmin/#/c/site-browser?&url=%s", encoder.encode( pageUrl ));
```

Comment:
Not sure if this should be configurable, but at least should be a constant easy to indentified